src/prototype_rust_sdk.rs

```diff
@@ -178,21 +180,17 @@ impl TestRustWorker {
 
 #[derive(Debug)]
 enum UnblockEvent {
-    Timer(String),
-    Activity(String, Box<ActivityResult>),
-    WorkflowStart(String, Box<ChildWorkflowStartStatus>),
-    WorkflowComplete(String, Box<ChildWorkflowResult>),
+    Timer(u32),
+    Activity(u32, Box<ActivityResult>),
+    WorkflowStart(u32, Box<ChildWorkflowStartStatus>),
+    WorkflowComplete(u32, Box<ChildWorkflowResult>),
 }
 
 #[derive(derive_more::From)]
 #[allow(clippy::large_enum_variant)]
 enum RustWfCmd {
     #[from(ignore)]
-    CancelTimer(String),
-    #[from(ignore)]
-    CancelActivity(String),
-    #[from(ignore)]
-    CancelChildWorkflow(String),
+    Cancel(CommandID),
```

Comment:
I don't really love this. It allows the construction of cancel commands that shouldn't be allowed to exist, like later on where child workflow completes can't be cancelled.

Reply:
We just make sure never to use that CommandID but I can make a narrower type with `impl From<CancellableID> for CommandID`

Reply:
Yeah what I'm getting at is this line shouldn't need to exist:
https://github.com/temporalio/sdk-core/pull/173/files#diff-0343a18509c554e2669813790fbe84d92fdea82f88d373bdeb5d69576148aac4R225